frontend/javascripts/admin/dataset/dataset_upload_view.tsx

```diff
@@ -890,6 +1014,29 @@ class DatasetUploadView extends React.Component<PropsWithFormAndRouter, State> {
                     return wkwFiles.length === 0 || imageFiles.length === 0;
                   }, "WKW files should not be mixed with image files."),
                 },
+                {
+                  validator: syncValidator(
+                    (files: FileWithPath[]) => {
+                      const { unfinishedUploadToContinue } = this.state;
+                      if (
+                        !unfinishedUploadToContinue ||
+                        unfinishedUploadToContinue.filePaths == null
+                      ) {
+                        return true;
+                      }
+                      const filePaths = files.map((file) => file.path || "");
+                      return (
+                        unfinishedUploadToContinue.filePaths.length === filePaths.length &&
+                        _.difference(unfinishedUploadToContinue.filePaths, filePaths).length === 0
+                      );
+                    },
+                    "The selected files do not match the files of the unfinished upload. Please select the same files as before." +
+                      unfinishedUploadToContinue?.filePaths !=
+                      null
+                      ? `The file names are ${unfinishedUploadToContinue?.filePaths?.join(", ")}.`
+                      : "",
```

Comment:
I think this is missing parentheses (the string is concated with the != null condition right now). I found this because I didn't see the proper error message during testing.

Reply:
Hmm strange, afaik this worked perfectly fine for me. But you are certainly correct, that the parentheses are needed due to operator precedence :thinking:

Thanks for catching this.